webview/src/plots/components/templatePlots/TemplatePlotsGrid.tsx

```diff
@@ -44,49 +46,47 @@ export const TemplatePlotsGrid: React.FC<TemplatePlotsGridProps> = ({
 }) => {
   const [order, setOrder] = useState<string[]>([])
   const [disabledDrag, setDisabledDrag] = useState('')
-  const nbVegaPanels = useRef(0)
+  const size = useSelector((state: PlotsState) => state.template.size)
 
-  useEffect(() => {
-    setOrder(entries.map(({ id }) => id))
-  }, [entries])
-
-  // eslint-disable-next-line sonarjs/cognitive-complexity
-  const layoutEffect = () => {
```

Comment:
With truncation, we somehow entered a race condition where this would be called before vega is ready. I found the `onNewView` method on the `<VegaLite />` component that gets called after the view is ready, which makes things much simpler and cleaner for us. That `onLayoutEffect` is also used elsewhere so I'll do a follow up after to use `onNewView` instead.